es-vue-base/src/lib-components/EsSlider.vue

```diff
@@ -0,0 +1,197 @@
+const gradient = `linear-gradient(104.98deg, ${colors.teal} -0.03%, ${colors.cyan} 50.02%, ${colors.indigo} 99.93%)`;
+
+export default {
+    name: 'EsSlider',
+    components: {
+        vueSlider,
+    },
+    props: {
+        /**
+         * Starting value of slider thumb
+         */
+        startingValue: {
+            type: Number,
+            required: true,
+        },
+        /**
+         * Array that contains valid steps
+         */
+        data: {
+            type: Array,
+            required: true,
+        },
+        /**
+         * Function that chooses where to add marks to the track
+         */
+        marks: {
+            type: Array,
+            required: true,
+        },
+        /**
+         * Aria Label for Dot
+         */
+        ariaLabel: {
+            type: String,
+            required: false,
+            default: 'Select a number',
+        },
+        /**
+         * Function that modifies tooltip value
+         */
+        tooltipFormatter: {
+            type: Function,
+            required: false,
+            default: (val) => val,
+        },
+        /**
+         * Choose when tooltip displays
+         */
+        tooltip: {
+            type: String,
+            required: false,
+            default: 'always',
+        },
+        /**
+         * Color variant
+         */
+        variant: {
+            type: String,
+            required: false,
+            default: 'primary',
+        },
+    },
+    data() {
+        return {
+            sliderValue: this.startingValue,
+            min: this.data[0],
+            max: this.data[this.data.length - 1],
+            colorComputed: variants[this.variant],
+            styles: {
+                dot: {
+                    border: `5px solid ${variants[this.variant]}`,
+                    background: colors.white,
+                },
+                rail: {
+                    height: '10px',
+                    'border-radius': '9px',
+                },
+                process: {
+                    height: '10px',
+                    background: gradient,
+                    'border-radius': '9px',
+                },
+                step: {
+                    display: 'none',
+                },
+            },
+        };
+    },
+    watch: {
+        startingValue: {
+            // Watch the prop and update the value to avoid mutation
+            handler(newVal) {
+                this.sliderValue = newVal > 0 ? newVal : this.startingValue;
+            },
+        },
+    },
+    methods: {
+        updateSliderValue(value) {
+            this.$emit('change', value);
+        },
+    },
+};
+</script>
+
+<style lang="scss">
+@import '~vue-slider-component/lib/theme/default';
+@import '~vue-slider-component/lib/styles/dot';
+@import '~vue-slider-component/lib/styles/mark';
+@import '~vue-slider-component/lib/styles/slider';
+</style>
+
+<style lang="scss" scoped>
```

Comment:
Could this be non-scoped?

Reply:
It could be, this also feels like some this could live in `es-bs-extends` as well. How do you think we should handle third party SCSS imports in general?